brave/src/main/java/brave/propagation/ExtraField.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.propagation;
+
+import brave.internal.Nullable;
+import brave.internal.PropagationFields;
+import java.util.Locale;
+
+import static brave.propagation.ExtraFieldPropagation.currentTraceContext;
+
+/**
+ * Defines a request-scoped field, usually but not always analogous to an HTTP header. Fields will
+ * be no-op unless {@link ExtraFieldPropagation} is configured.
+ *
+ * <p>For example, if you have a need to know the a specific request's country code, you can
+ * propagate it through the trace:
+ * <pre>{@code
+ * // Configure your extra field
+ * countryCode = ExtraField.create("country-code");
+ *
+ * // If configured and set, you can retrieve it later. All of the below result in the same tag:
+ * spanCustomizer.tag(countryCode.name(), countryCode.getValue());
+ * spanCustomizer.tag(countryCode.name(), countryCode.getValue(context));
+ * spanCustomizer.tag("country-code", ExtraField.getValue("country-code"));
+ *
+ * // You can also set or override the value similarly, which might be needed if a new request
```

Comment:
'for a new request' rather ? otherwise the sentence looked incomplete at first sight - though grammatically correct

Reply:
You can also update the value similarly, so that the new value will propagate downstream.

comment back if needed